### PR TITLE
Non assert 5 x

### PR DESCRIPTION
**Pull request by lisaoakley**
Backport to 5X_STABLE the pattern of building with asserts should only be driven by configure flag.

Included with this, we have a need for release candidates of 5X_STABLE built without asserts for performance testing.

### CODE REVIEW

**Comment by pivotal-issuemaster** [comment]
@lisaoakley Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/greenplum?repositoryId=greenplum-db/gpdb&pullRequestId=5163)!

[Click here](https://cla.pivotal.io/sync/greenplum?repositoryId=greenplum-db/gpdb&pullRequestId=5163) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/about) for frequently asked questions.

**Comment by pivotal-issuemaster** [comment]
@lisaoakley Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/greenplum?repositoryId=greenplum-db/gpdb&pullRequestId=5163)!

[Click here](https://cla.pivotal.io/sync/greenplum?repositoryId=greenplum-db/gpdb&pullRequestId=5163) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/about) for frequently asked questions.

File: configure [hunk 1]
```diff
@@ -5990,8 +5990,6 @@ fi
 # Enable assert checks
 #
 
-pgac_args="$pgac_args enable_cassert"
```
Comment (kmacoskey) [inline configure]: Looking at why this changed in gpdb6 and how it relates to autoconf updates, I still don't fully understand why this change is necessary.

Comment (kmacoskey) [reply configure]: I ran a quick test locally and with this line in place and the behavior is as expected. The default is that asserts are disabled and it only gets enabled when you pass the flag on configure. So there should be no reason for us to remove this.

**Comment by kmacoskey** [comment]
The was an incorrect assumption at the start of this that the 5X_STABLE pipeline was building with asserts. Because the pipeline is already building without asserts, we don't currently need to add any additional jobs in place to make sure we get a release candidate being built without asserts.

In the future if the 5X_STABLE pipeline were to be set to build with asserts then we would push back and say there needs to then be two pipelines. This would still satisfy the need to have 5X being compiled without asserts that can be used for release candidates.

Those two things put together means we can drop these changes.
